### DISCUSSION TITLE
Parsing algolia response and use it as a payload to make an additional call to another source (Not Algolia index) and bind that result to OOTB instantSearch widget

### DISCUSSION THREAD

**Discussion by Sbhatthorizontal** [algolia/instantsearch#5067] [Ideas]
Hello Algolia,

What i am looking to implement is that by using React hooks instantSearch, i want to get the response from Algolia and parse it. I want to use that response as a payload for my next call to OrderCloud to fetch some additional details which i can bind into OOTB InstantSearch Widget to display on UI.

**Comment by francoischalifour** [comment]
You could [implement a search client proxy](https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/#implementing-a-proxy) or [use a custom backend](https://www.algolia.com/doc/guides/building-search-ui/going-further/backend-search/in-depth/backend-instantsearch/js/).

**Reply by Sbhatthorizontal** [reply]
Thanks @francoischalifour for your time. I have gone through above links but i am unable to find anything which can solve my purpose.  Below is the UseCase which i am trying to implement.

1) Implement React InstantSearch.
2) Get the response from Algolia using instantsearch.
3) Extract few details from the response.
4) Create a payload (query parameter) from those extracted data.
5) Make a call to OrderCloud with new payload.
6) Bind the response of Ordercloud into OOTB instantsearch widgets (hits template)

So to summarize this, i need to inject and additional call to Ordercloud before displaying search results into UI as we need to show some additional details which is available in Ordercloud and not on Algolia.

Can you please help me with that?

Thank you so much for your time and help.

**Reply by francoischalifour** [reply]
It seems like you could patch the `search()` method of the search client to run its original behavior, plus target the OrderCloud endpoint, and merge the OrderCloud response into the hits.

```js
const originalClient = algoliasearch('YOUR_APP_ID', 'YOUR_API_KEY');

const searchClient = {
  ...originalClient,
  search(requests) {
    return originalClient.search(requests).then((response) => {
      // call your endpoint and merge the hits
    });
  },
};
```
